kombu/transport/SQS.py

```diff
@@ -569,11 +569,14 @@ def _get_bulk_async(self, queue,
         return callback
 
     def _get_async(self, queue, count=1, callback=None):
-        q = self._new_queue(queue)
+        q_url = self._new_queue(queue)
         qname = self.canonical_queue_name(queue)
         return self._get_from_sqs(
-            qname, count=count, connection=self.asynsqs(queue=qname),
-            callback=transform(self._on_messages_ready, callback, q, queue),
+            queue_name=qname, queue_url=q_url, count=count,
+            connection=self.asynsqs(queue=qname),
+            callback=transform(
```

Comment:
this art will need appropriate test coverage

Reply:
Added in the latest commit